src/equations/shallow_water_exner_1d.jl

```diff
@@ -0,0 +1,618 @@
+                                                                                            S
+                                                                                            }
+    (; gravity, rho_f, rho_s, porosity_inv) = equations
+    (; m_1, m_2, m_3, k_1, k_2, k_3, theta_c, d_s) = equations.sediment_model
+
+    theta = rho_f * abs(shear_stress(u, equations)) / (gravity * (rho_s - rho_f) * d_s)  # Shields stress
+
+    Q = d_s * sqrt(gravity * (rho_s / rho_f - 1.0) * d_s) # Characteristic discharge
+
+    return porosity_inv * Q * sign(theta) * k_1 * theta^m_1 *
+           (max(theta - k_2 * theta_c, 0.0))^m_2 *
+           (max(sqrt(theta) - k_3 * sqrt(theta_c), 0.0))^m_3
+end
+
+# Compute the sediment discharge for the Grass model 
+@inline function q_s(u,
+                     equations::ShallowWaterExnerEquations1D{T, S, GrassModel{T}}) where {
+                                                                                          T,
+                                                                                          S
+                                                                                          }
+    (; porosity_inv, sediment_model) = equations
+    return porosity_inv * sediment_model.A_g * velocity(u, equations)^3
+end
+
+# Shear stress formulation using a coefficient to take into account different friction models
+@inline function shear_stress(u, equations::ShallowWaterExnerEquations1D)
+    v = velocity(u, equations)
+    return equations.gravity * shear_stress_coefficient(u, equations.friction) * v *
+           abs(v)
+end
+
+# Model dependent shear stress coefficient
+@inline function shear_stress_coefficient(u, friction::ManningFriction)
+    h, _, _ = u
+    return friction.n^2 / h^(1 / 3)
+end
+
+# Convert conservative variables to primitive
+@inline function Trixi.cons2prim(u, equations::ShallowWaterExnerEquations1D)
+    h, _, h_b = u
+
+    H = h + h_b
+    v = velocity(u, equations)
+
+    return SVector(H, v, h_b)
+end
+
+# Convert conservative variables to entropy variables
+@inline function Trixi.cons2entropy(u, equations::ShallowWaterExnerEquations1D)
+    h, _, h_b = u
+    (; gravity, r) = equations
+
+    v = velocity(u, equations)
+
+    w1 = r * (gravity * (h + h_b) - 0.5 * v^2)
+    w2 = r * v
+    w3 = gravity * (r * h + h_b)
+
+    return SVector(w1, w2, w3)
+end
+
+# Convert primitive to conservative variables
+@inline function Trixi.prim2cons(prim, equations::ShallowWaterExnerEquations1D)
+    H, v, h_b = prim
+
+    h = H - h_b
+    hv = h * v
+
+    return SVector(h, hv, h_b)
+end
+
+@inline function Trixi.waterheight(u, equations::ShallowWaterExnerEquations1D)
+    return u[1]
+end
+
+# Indicator variable used for the shock capturing in `IndicatorHennemannGassnerShallowWater`
+@inline function water_sediment_height(u, equations::ShallowWaterExnerEquations1D)
+    return equations.gravity * u[1] * u[3]
+end
+
+# Entropy function
```

Comment:
```suggestion
# Entropy function that is the total energy of the Shallow Water-Exner model.
```

Reply:
I would suggest to name it "Mathematical entropy function" as the kinetic energy of the sediment layer is missing in the total energy.

Reply:
Done in https://github.com/trixi-framework/TrixiShallowWater.jl/pull/51/commits/f73f6e980707b325869773513dfc0f07da633710